### PR TITLE
Add an api endpoint for receiving requests to retry

### PR DESCRIPTION
**Pull request by justinvdm**
Will make use of the functionality added in #3.

### CODE REVIEW

**Comment by justinvdm** [comment]
Depends on #3, otherwise ready for review.

**Comment by hodgestar** [comment]
Looks good to me on a first pass, but I should probably re-review after the merge of #3 in case I missed anything small.

**Comment by justinvdm** [comment]
I think this is reviewable now.

File: vumi_http_retry/tests/test_app.py [hunk 1]
```diff
@@ -8,21 +9,30 @@
 from twisted.internet.defer import inlineCallbacks
 
 from vumi_http_retry.app import VumiHttpRetryServer
+from vumi_http_retry.retries import requests_key
+from vumi_http_retry.tests.redis import zitems, delete
 
 
 class TestVumiHttpRetryServer(TestCase):
+    timeout = 1
```
Comment (rudigiesler) [inline vumi_http_retry/tests/test_app.py]: This should probably be removed? Unless there's a reason for it.

Comment (justinvdm) [inline vumi_http_retry/tests/test_app.py]: Woops, stray debugging line, thanks.

File: vumi_http_retry/app.py [hunk 2]
```diff
@@ -29,6 +54,22 @@ def respond(cls, req, data, code=http.OK):
...
+
+    @app.route('/requests/', methods=['POST'])
+    @inlineCallbacks
+    def route_requests(self, req):
+        # TODO validation
+        # TODO rate limiting
+        data = json.loads(req.content.read())
+
+        yield add_request(self.redis, self.config.redis_prefix, {
+            'owner_id': req.getHeader('x-owner-id'),
```
Comment (hodgestar) [inline vumi_http_retry/app.py:69]: We should probably return an error if there isn't an owner id?

File: vumi_http_retry/tests/test_app.py [hunk 3]
```diff
@@ -30,8 +38,46 @@ def start_server(self):
...
+                'url': 'http://www.example.org',
+                'method': 'GET',
+            }
+        }, headers={'X-Owner-ID': '1234'})
+
+        self.assertEqual(resp.code, http.OK)
+        self.assertEqual((yield resp.content()), json.dumps({}))
+
+        self.assertEqual((yield zitems(self.app.redis, k)), [
+            (10 + 30, {
```
Comment (hodgestar) [inline vumi_http_retry/tests/test_app.py]: Should the `10` here be `self.time`?

**Comment by hodgestar** [comment]
Two minor comments, otherwise looks good.

File: vumi_http_retry/app.py (hunk 2 above)
Comment (justinvdm) [inline vumi_http_retry/app.py:69]: Yup, was assuming that's part of the validation TODO.

Comment (hodgestar) [inline vumi_http_retry/app.py:69]: Cool.

Comment (justinvdm) [inline vumi_http_retry/app.py:69]: Could also add it now, if that would be better? If you don't have a preference, I'd like to do that with the other validation stuff.

**Comment by hodgestar** [comment]
:+1:

File: vumi_http_retry/app.py (hunk 2 above)
Comment (hodgestar) [inline vumi_http_retry/app.py:69]: With other validation stuff is perfect.

**Pull request merged**